frontend/public/components/routes.tsx

```diff
@@ -186,13 +188,48 @@ const calcTrafficPercentage = (weight: number, route: any) => {
   return `${percentage.toFixed(1)}%`;
 };
 
+const getIngressStatusForHost = (hostname, ingresses): IngressStatusProps => {
+  return _.find(ingresses, {host: hostname}) as IngressStatusProps;
+};
+
+const showCustomRouteHelp = (ingress, annotations) => {
+  if (!ingress || !_.some(ingress.conditions, {type: 'Admitted', status: 'True'})) {
+    return false;
+  }
+
+  if (_.get(annotations, 'openshift.io/host.generated') === 'true') {
+    return false;
+  }
+
+  if (!ingress.host || !ingress.routerCanonicalHostname) {
+    return false;
+  }
+
+  return true;
+};
+
 const RouteTargetRow = ({route, target}) => <tr>
   <td><ResourceLink kind={target.kind} name={target.name} namespace={route.metadata.namespace} title={target.name} /></td>
   <td>{target.weight}</td>
   <td>{calcTrafficPercentage(target.weight, route)}</td>
 </tr>;
 
-const RouteIngressStatus: React.SFC<RouteIngressStatusProps> = ({ingresses}) =>
+const CustomRouteHelp: React.SFC<CustomRouteHelpProps> = ({host, routerCanonicalHostname}) =>
+  <Popover
+    headerContent={<span>Custom route</span>}
```

Comment:
Unnecessary span?

Reply:
No, needs to be a ReactElement.

Reply:
Then I'd suggest `<React.Fragment>`.

In Patternfly, we might look at allowing this to be any `React.ReactNode`, which includes strings, instead of forcing `React.ReactElement` / `JSX.Element`.

Reply:
btw, this should probably be `Custom Route` to match titles we use elsewhere (such as dialogs)

Reply:
Ok, fixed both issues